eclipse-collections-api/src/main/java/org/eclipse/collections/api/ordered/OrderedIterable.java

```diff
@@ -325,6 +326,26 @@
     @Override
     <S, R extends Collection<Pair<T, S>>> R zip(Iterable<S> that, R target);
 
+    /**
+     * Returns a {@code OrderedIterable} formed from this {@code OrderedIterable} and another {@code Iterable} by
+     * applying the {@code Function2}. The second {@code Iterable} should also be ordered.
```

Comment:
by applying the {@code Function2} **to each corresponding pair**

I think the description on the `return` Javadoc tag is clear about that, but the summary can be more clear too.